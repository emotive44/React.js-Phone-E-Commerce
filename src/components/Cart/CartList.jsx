import React from 'react';
import CartItem from './CartItem';

export default function CartList({ data, cart }) {
    if(!cart) { return null }
    return (
        <div className="container-fluid">
            {
                cart.map(item => {
                    return <CartItem 
                        key={item.id}
                        item={item}
                        data={data}
                    />
                })
            }
        </div>
    )
}