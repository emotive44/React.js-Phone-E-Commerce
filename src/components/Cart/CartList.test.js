import React from 'react';
import CartList from './CartList';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    Link: 'Link'
}));

describe('CartList component', () => {
    test('should reder null if cart is undefined', () => {
        const component = renderer.create(<CartList />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})