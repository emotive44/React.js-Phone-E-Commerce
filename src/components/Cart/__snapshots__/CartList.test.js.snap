// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartList component should reder null if cart is undefined 1`] = `null`;
